fairtools/MCConfigurator/FairXmlVMCConfig.cxx

```diff
@@ -0,0 +1,429 @@
+/********************************************************************************
+ *    Copyright (C) 2022 GSI Helmholtzzentrum fuer Schwerionenforschung GmbH    *
+ *                                                                              *
+ *              This software is distributed under the terms of the             *
+ *              GNU Lesser General Public Licence (LGPL) version 3,             *
+ *                  copied verbatim in the file "LICENSE"                       *
+ ********************************************************************************/
+// -------------------------------------------------------------------------
+// -----               FairXmlVMCConfigurce file                       -----
+// -----            Created 2019.02.19  by R. Karabowicz               -----
+// -------------------------------------------------------------------------
+#include "FairXmlVMCConfig.h"
+
+#include "FairFastSimRunConfiguration.h"
+#include "FairLogger.h"
+#include "FairRunSim.h"
+#include "FairSink.h"   // for FairSink
+
+#include <Rtypes.h>
+#include <TGeant3.h>
+#include <TGeant3TGeo.h>
+#include <TGeant4.h>
+#include <TObjString.h>   // for TObjString
+#include <TObject.h>      // for TObject, TObject::kSingleKey
+#include <TString.h>
+#include <TSystem.h>      // for TSystem, gSystem
+#include <TVirtualMC.h>   // for TVirtualMC
+#include <cstdlib>        // for getenv
+#include <cstring>        // for strcmp, strncmp
+#include <ostream>        // for operator<<, ostringstream
+#include <string>         // for string, basic_string, cha...
+#include <vector>         // for vector
+
+FairXmlVMCConfig::FairXmlVMCConfig()
+    : FairGenericVMCConfig()
+    , fMCEngine("")
+    , fXmlEngine(new TXMLEngine())
+{
+    UsePostInitConfig(false);
+}
+
+void FairXmlVMCConfig::Setup(const char* mcEngine)
+{
+    fMCEngine = mcEngine;
+    if (!((strcmp(mcEngine, "TGeant4") == 0) || (strcmp(mcEngine, "TGeant3") == 0))) {
+        LOG(fatal) << "FairXmlVMCConfig::Setup() Engine \"" << mcEngine << "\" unknown!";
+    }
+
+    string xmlFileName = ObtainXmlFileName(mcEngine);
+
+    // Parse xml file
+    // Only file with restricted xml syntax are supported
+    XMLDocPointer_t xmldoc = fXmlEngine->ParseFile(xmlFileName.data());
+    if (!xmldoc)
+        return;
```

Comment:
How does this work out? If parsing failed, we just silently skip the rest without any error? 🤯